### ISSUE TITLE
Unable to train good model on custom dataset.

### ISSUE THREAD

**Issue by mansi-aggarwal-2504** [matterport/Mask_RCNN#2669]
I have been trying to train mask rcnn on custom dataset but I am not getting good results on my val or test dataset.
I also tried applying augmentations but that didn't work either. (see [issue](https://github.com/matterport/Mask_RCNN/issues/2663))

This time, I tried to slightly increase the learning rate, and did some modifications to my data. Both train and val images are 256x256 whereas earlier the train data images were 512x512 in size. 

This was my learning curve:
![MicrosoftTeams-image (25)](https://user-images.githubusercontent.com/60866597/129516173-f01888a1-4c71-47b8-b7e6-a93d9f1eb2ce.png)

I am not sure what to do now. The model just doesn't seem to perform well on val data under any circumstances. Any suggestions on what I might be doing wrong or what I could try?

**Comment by mansi-aggarwal-2504** [comment]
Update: I checked the masks on val and a new test dataset, got the bounding boxes also. The confidence threshold was set 0.7 for inference. The masks are pretty decent even for such a bad learning curve. I used the model checkpoint at 10th epoch since I saved models at steps of 10 and the model was immediately overfitting, I took the first checkpoint.

I don't understand this behaviour though. Can someone help?
